File: app/src/main/java/com/playgilround/schedule/client/tutorial/TutorialPresenter.java
package com.playgilround.schedule.client.tutorial;

import android.content.Context;

import com.playgilround.schedule.client.data.repository.UsersRepository;

/**
 * 19-02-20
 * Tutorial 관련 데이터 처리 Presenter
 */
public class TutorialPresenter implements TutorialContract.Presenter {

    private final Context mContext;
    private final TutorialContract.View mView;
    private final UsersRepository mUsersRepository;

    TutorialPresenter(Context context, TutorialContract.View view, UsersRepository usersRepository) {
        mContext = context;
        mView = view;
        mView.setPresenter(this);
        mUsersRepository = usersRepository;
    }

    @Override
    public void start() {

    }

    @Override
    public void checkLogin() {
        if (mUsersRepository.getCurrentUser(mContext) != null) {
            mView.skipTutorial();
        }
    }
}

File: app/src/main/java/com/playgilround/schedule/client/tutorial/TutorialPresenter.kt
package com.playgilround.schedule.client.tutorial

import android.content.Context
import com.playgilround.schedule.client.ScheduleApplication
import com.playgilround.schedule.client.data.repository.UsersRepository
import javax.inject.Inject

/**
 * 19-08-07 cho
 */
class TutorialPresenter constructor(private val mContext: Context, private val mView: TutorialContract.View): TutorialContract.Presenter {

    @Inject
    internal lateinit var mUsersRepository: UsersRepository

    init {
        mView.setPresenter(this)
        (mContext.applicationContext as ScheduleApplication).appComponent.tutorialInject(this)
    }

    override fun start() {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun checkLogin() {
        if (mUsersRepository.getCurrentUser(mContext) != null) mView.skipTutorial()
    }
}